### PR TITLE
[WIP] Cleanup and flatten imports

### PR DESCRIPTION
**Pull request by yashgorana**
## Description

Part of https://github.com/OpenMined/Heartbeat/issues/284

## Affected Dependencies
List any dependencies that are required for this change.

## How has this been tested?
- Describe the tests that you ran to verify your changes.
- Provide instructions so we can reproduce.
- List any relevant details for your test configuration.

## Checklist
- [ ] I have followed the [Contribution Guidelines](https://github.com/OpenMined/.github/blob/master/CONTRIBUTING.md) and [Code of Conduct](https://github.com/OpenMined/.github/blob/master/CODE_OF_CONDUCT.md)
- [ ] I have commented my code following the [OpenMined Styleguide](https://github.com/OpenMined/.github/blob/master/STYLEGUIDE.md)
- [ ] I have labeled this PR with the relevant [Type labels](https://github.com/OpenMined/.github/labels?q=Type%3A)
- [ ] My changes are covered by tests

### CODE REVIEW

File: ruff.toml [hunk 1]
```diff
@@ -5,3 +5,6 @@ line-length = 120
 select = ["E", "F", "B"]
 
 ignore = ["B904", "B905"]
+
+[per-file-ignores]
```
Comment (yashgorana) [inline ruff.toml:9]: a neat way to avoid adding `#noqa: F401` to all `__init__.py` files

**Comment by yashgorana** [comment]
Instantly detects a whole bunch of circular and incorrect imports - especially between `client`, `node` & `service`. I have a couple of ideas to resolve this, but need to discuss in detail. Also a quick heads up - the change can get quite big =)

**Comment by kiendang** [comment]
> Instantly detects a whole bunch of circular and incorrect imports - especially between client, node & service. I have a couple of ideas to resolve this, but need to discuss in detail.

Would be great if we can resolve this. Currently there are a bunch of places where we have to do lazy import inside function body to work around this, which also means we have to give up on typing.

**Comment by yashgorana** [comment]
After discussing with Madhava, this PR is being targeted for 0.8.2 and will be put on the back burner for this week.

**Comment by madhavajay** [comment]
@shubham3121 @yashgorana do you think we can refactor and do this during cool down for 0.8.2 or should we abandon it?

**Pull request closed without merging**